### PR TITLE
Add server start remarks + UE pass to Servers topic

### LINKED ISSUE #5673: Consider adding `dotnet run` on aspnet core apps starts Kestrel
Might want to mention dev environment too.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel and maybe https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #5673 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/index?branch=pr-en-us-5686)

* UE pass
* Add a section on how the server starts calling out `dotnet run`

### CODE REVIEW

**Comment by guardrex** [comment]
@Tratcher Do u have a minute to take a look at one new section here? I added a bit on how the server starts. I'm not trying to get too deep in the weeds with it. I am trying to meet @Rick-Anderson's ask to surface how the server starts, particularly that `dotnet run` starts the server from a command prompt.

The section is at the bottom, and this link will take you right to it in the review topic:

https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/index?branch=pr-en-us-5686#server-startup

You can ignore the rest of the updates here ... they're just UE updates (removing "you/your," xref-style links, phrasing, guidelines adherence, etc.).

File: aspnetcore/fundamentals/servers/index.md [hunk 1]
```diff
@@ -14,126 +15,104 @@ uid: fundamentals/servers/index
...
 
-## Next steps
+## Server startup
 
-For more information, see the following resources:
+When using [Visual Studio](https://www.visualstudio.com/vs/) or [Visual Studio Code](https://code.visualstudio.com/), the server is launched when the app is started by the Integrated Development Environment (IDE). In Visual Studio, [IIS Express](/iis/extensions/introduction-to-iis-express/iis-express-overview) is used with the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) to launch the app and server. In Visual Studio Code, the app and server are started by [Omnisharp](https://github.com/OmniSharp/omnisharp-vscode).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/servers/index.md]: VS for Mac.
Not sure how that works, but the IDE does start Kestrel - I think.

Comment (guardrex) [reply aspnetcore/fundamentals/servers/index.md]: Yeah ... good point. I'll have to find out about VS for Mac and interject a little something there about how that works.

**Comment by guardrex** [comment]
I'm piecing together a (barely) educated guess based on online resources. Those remarks in the first paragraph require careful consideration.

**Comment by Rick-Anderson** [comment]
@VanKichline can you comment on web server VS for Mac launches?

File: aspnetcore/fundamentals/servers/index.md [hunk 2]
```diff
@@ -14,126 +15,104 @@ uid: fundamentals/servers/index
...
 
-## Next steps
+## Server startup
 
-For more information, see the following resources:
+When using [Visual Studio](https://www.visualstudio.com/vs/), [Visual Studio for Mac](https://www.visualstudio.com/vs/mac/), or [Visual Studio Code](https://code.visualstudio.com/), the server is launched when the app is started by the Integrated Development Environment (IDE). In Visual Studio on Windows, [IIS Express](/iis/extensions/introduction-to-iis-express/iis-express-overview) is used with the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) to launch the app and server. In Visual Studio Code, the app and server are started by [Omnisharp](https://github.com/OmniSharp/omnisharp-vscode), which activates the CoreCLR debugger. Using Visual Studio for Mac, the app and server are started by the [Mono Soft-Mode Debugger](http://www.mono-project.com/docs/advanced/runtime/docs/soft-debugger/).
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/index.md]: In VS you can choose your launch profile to launch with IIS Express or with the console.

File: aspnetcore/fundamentals/servers/index.md [hunk 3]
```diff
@@ -14,126 +15,104 @@ uid: fundamentals/servers/index
...
 
-# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+When launching an app from a command prompt in the project's folder, [dotnet run](/dotnet/core/tools/dotnet-run) launches the app and server. The configuration is specified by the `-c|--configuration` option, which is set to either `Debug` (default) or `Release`. If launch profiles are present in a *launchSettings.json* file, use the `--launch-profile <NAME>` option to set the launch profile (for example, `Development` or `Production`). For more information, see the [dotnet run](/dotnet/core/tools/dotnet-run) and [.NET Core distribution packaging](/dotnet/core/build/distribution-packaging) topics.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/index.md]: Kestrel or HttpSys can be launched this way.

File: aspnetcore/fundamentals/servers/index.md [hunk 4]
```diff
@@ -14,126 +15,104 @@ uid: fundamentals/servers/index
...
+## ASP.NET Core server infrastructure
 
-The [`IApplicationBuilder`](/aspnet/core/api/microsoft.aspnetcore.builder.iapplicationbuilder) available in the `Startup` class `Configure` method exposes the `ServerFeatures` property of type [`IFeatureCollection`](/aspnet/core/api/microsoft.aspnetcore.http.features.ifeaturecollection). Kestrel and WebListener both expose only a single feature, [`IServerAddressesFeature`](/aspnet/core/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature), but different server implementations may expose additional functionality.
+The [IApplicationBuilder](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder) available in the `Startup.Configure` method exposes the [ServerFeatures](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder.serverfeatures) property of type [IFeatureCollection](/dotnet/api/microsoft.aspnetcore.http.features.ifeaturecollection). Kestrel and WebListener only expose a single feature each, [IServerAddressesFeature](/dotnet/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature), but different server implementations may expose additional functionality.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/index.md]: HttpSys

File: aspnetcore/fundamentals/servers/index.md [hunk 5]
```diff
@@ -105,14 +105,14 @@ If the built-in servers don't meet the app's requirements, a custom server imple
...
+When using [Visual Studio](https://www.visualstudio.com/vs/), [Visual Studio for Mac](https://www.visualstudio.com/vs/mac/), or [Visual Studio Code](https://code.visualstudio.com/), the server is launched when the app is started by the Integrated Development Environment (IDE). In Visual Studio on Windows, launch profiles can be used to start the app and server with either [IIS Express](/iis/extensions/introduction-to-iis-express/iis-express-overview)/[ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) or the [Windows console](/windows/console/consoles). In Visual Studio Code, the app and server are started by [Omnisharp](https://github.com/OmniSharp/omnisharp-vscode), which activates the CoreCLR debugger. Using Visual Studio for Mac, the app and server are started by the [Mono Soft-Mode Debugger](http://www.mono-project.com/docs/advanced/runtime/docs/soft-debugger/).
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/index.md]: How is linking to console docs helpful here? This is more like the dotnet run scenario bellow.

**Comment by guardrex** [comment]
@Rick-Anderson Ready for another look. Note one thing tho ... the sentence ...

> In Visual Studio on Windows, launch profiles can be used to start the app and server with either [IIS Express](#)/[ASP.NET Core Module](#) or the [Windows console](#).

I tried breaking that out. For example ...

In Visual Studio on Windows, launch profiles can be used to start the app and server with:

* [IIS Express](#)/[ASP.NET Core Module](#)
* [Windows console](#)

I tried several formats using a list ... with and without sentence completion (and periods) .. with and without a definite article on "Windows console." It seems to work better as an ordinary sentence. As a sentence, it can make it clear that the launch profiles are **either** going to allow use of one approach **or** the other. Just mentioning in case u wonder why I used a sentence over a list.

File: aspnetcore/fundamentals/servers/index.md (hunk 5 above)
Comment (guardrex) [reply aspnetcore/fundamentals/servers/index.md]: Definition only. I went a bit overboard. I'll remove it.

**Comment by guardrex** [comment]
@Rick-Anderson Do you want to look this one over? ... or see if `@scottaddie` is free?

**Pull request merged**